nncf/quantization/algorithms/min_max/algorithm.py

```diff
@@ -162,22 +170,73 @@ def __init__(
         }
 
         # preset definition
-        if preset is None:
+        if self._preset is None:
             if model_type == ModelType.TRANSFORMER:
-                preset = QuantizationPreset.MIXED
+                self._preset = QuantizationPreset.MIXED
             else:
-                preset = QuantizationPreset.PERFORMANCE
+                self._preset = QuantizationPreset.PERFORMANCE
 
+        if self._mode is not None:
+            self._redefine_defaults_based_on_mode()
         # Calculates global quantizer constraints
         self._global_quantizer_constraints = {}
         for quantizer_group in QuantizerGroup:
             self._global_quantizer_constraints[quantizer_group] = self._get_quantizer_constraints(
-                quantizer_group, preset, self._quantization_params[quantizer_group]
+                quantizer_group, self._preset, self._quantization_params[quantizer_group]
             )
 
         self._reset_cache()
         self._algorithm_key = f"MMQ_{hash(self)}"
 
+    def _redefine_defaults_based_on_mode(self):
```

Comment:
I think that we shouldn't do it for INT8 quantization, since it would affect user experience. If the user provided the wrong configuration, we should say about it, and not change the parameters as we want.
In this case, the existence of the `_redefine_defaults_based_on_mode` is really weird. But since we do workarounds for FP8, there is no other way to change it rapidly.

Reply:
Let's discuss it offline